dvc/state.py

```diff
@@ -110,10 +111,13 @@ def __init__(self, repo, config):
 
         self.state_file = os.path.join(self.dvc_dir, self.STATE_FILE)
 
+        self.lock = Lock(self.dvc_dir, "state.lock")
```

Comment:
are you sure we actually need to lock SQLLite artificially? https://www.sqlite.org/lockingv3.html

it supports ACID transactions in some capacity, so you should be able to update DB in parallel.

Reply:
We do, otherwise we will get I/O errors, since we mostly do write operations(even when reading some checksum entry, we are updating it's timestamp).

Reply:
https://sqlite.org/faq.html#q5

> When any process wants to write, it must lock the entire database file for the duration of its update.